core/forkchoice.go

```diff
@@ -114,7 +114,9 @@ func (f *ForkChoice) ReorgNeeded(current *types.Header, extern *types.Header) (b
 		if f.preserve != nil {
 			currentPreserve, externPreserve = f.preserve(current), f.preserve(extern)
 		}
-		reorg = !currentPreserve && (externPreserve || f.rand.Float64() < 0.5)
+		reorg = !currentPreserve && (externPreserve ||
+			extern.Time < current.Time ||
+			extern.Time == current.Time && f.rand.Float64() < 0.5)
```

Comment:
Come from postmortem sharing, just feel it might be better to write as 
reorg = !currentPreserve && (externPreserve ||
			extern.Time < current.Time ||
			(extern.Time == current.Time && f.rand.Float64() < 0.5)) , adding a pair of parentheses for the last && to make the logic clearer. Not sure about the context, so please correct me if I misunderstand.

Reply:
> Come from postmortem sharing, just feel it might be better to write as reorg = !currentPreserve && (externPreserve || extern.Time < current.Time || (extern.Time == current.Time && f.rand.Float64() < 0.5)) , adding a pair of parentheses for the last && to make the logic clearer. Not sure about the context, so please correct me if I misunderstand.

thx for review!

the operator `&&` has a higher prority than `||`.
If I write 117~119 into one line, I will follow your code to add a pair of parentheses.
here, I spilt 3 conditions into 3 lines, so It's also clear to me.

Reply:
> Come from postmortem sharing, just feel it might be better to write as reorg = !currentPreserve && (externPreserve || extern.Time < current.Time || (extern.Time == current.Time && f.rand.Float64() < 0.5)) , adding a pair of parentheses for the last && to make the logic clearer. Not sure about the context, so please correct me if I misunderstand.

thx for review!

the operator && has a higher prority than ||.
If I write 117~119 into one line, I will follow your code to add a pair of parentheses.
here, I spilt 3 conditions into 3 lines, so It's also clear to me.